File: Solutions/17675/c++/program.cpp
// Copy from Quera

// In the name of God

#include <iostream>
using namespace std;

int fib(int n)
{
    if (n <= 1)
        return 1;
    return fib(n - 1) + fib(n - 2);
}

int main()
{
    int n;
    cin >> n;
    for (int i = 1; i <= n; i++)
    {
        int j = 1;
        while (fib(j) < i)
            j++;
        if (fib(j) == i)
            cout << '+';
        else
            cout << '-';
    }
    cout << '\n';
    return 0;
}

File: Solutions/17675/java/program.java
//Copy from Quera

import java.util.Scanner;

public class Main
{

    public static void main(String[] args)
    {
        Scanner scanner=new Scanner(System.in);
        int n=scanner.nextInt();
        boolean []isFib=new boolean[101];
        for(int i=1;;i++)
        {
            int x=fib(i);
            if(x>n)break;
            isFib[x]=true;
        }
        for(int i=1;i<=n;i++)
        {
            if(isFib[i]) System.out.print("+");
            else System.out.print("-");
        }

    }

    private static int fib(int x)
    {
        if(x==1)return 1;
        if(x==2)return 2;
        return fib(x-1)+fib(x-2);
    }


}